File: Linked List/1. LinkedList/4. LinkedList Search/LinkedListSearch.java
import java.util.Random;

class Node
{
    int value;
    Node next;

    public Node()
    {
        this.next = null;
    }

    public Node(int value)
    {
        this.value = value;
        this.next = null;
    }
}

public class LinkedListSearch
{
    private Node head;

    public LinkedListSearch()
    {
        head = null;
    }

    public boolean isEmpty()
    {
        if (head == null)
            return true;
        else
            return false;
    }

    public void printList()
    {
        if (isEmpty())
            System.out.println("LinkedList is Empty");
        
        Node temp = head;

        while (temp != null)
        {
            System.out.print(temp.value + "->");
            temp = temp.next;
        }
        System.out.println("null");
    }

    public void insertAtTail(int value)
    {
        Node newNode = new Node(value);

        if (isEmpty())
        {
            newNode.next = head;
            head = newNode;
            System.out.println(value + " has been inserted");
            return;
        }
        Node temp = head;
        while (temp.next != null)
        {
            temp = temp.next;
        }
        temp.next = newNode;
        System.out.println(value + " has been inserted.");
    }

    public boolean search(int value)
    {
        Node  temp = head;

        while (temp != null)
        {
            if (temp.value == value)
                return true;
            temp = temp.next;
        }
        temp = null;
        return false;

    }

    public static void main(String args[])
    {
        LinkedListSearch list = new LinkedListSearch();

        for (int i = 0; i < 10; i++)
        {
            list.insertAtTail(i);
            list.printList();
        }

        Random rand = new Random();
        boolean exist = false;

        for (int i = 0; i < 5; i++)
        {
            int randInt = rand.nextInt(20);
            exist = list.search(randInt);

            if (exist)
                System.out.println(randInt + " exists.");
            else
                System.out.println(randInt + " does not exist.");
        }
    }
}

File: Linked List/1. LinkedList/4. LinkedList Search/LinkedListSearch.cpp
#include <iostream>
using namespace std;

class Node
{
public:
    int data;
    Node* nextElement;
    Node(/* args */)
    {
        nextElement = nullptr;
    }
};

class LinkedList
{
    private:
        Node* head;
    
    public:
        LinkedList()
        {
            head = nullptr;
        }

        bool isEmpty()
        {
            if (head == nullptr)
                return true;
            else
            {
                return false;
            }
        }

        // Print the LinkedList
        bool printList()
        {
            if (isEmpty())
            {
                cout << "List is Empty" << endl;
                return false;
            }

            Node* temp = head;
            cout << "List: ";

            while (temp != nullptr)
            {
                cout << temp->data << "->" ;
                temp = temp->nextElement;
            }
            cout << "null" << endl;
            return true;
        }

        // insert a new node as the first element of the list
        void insertAtHead(int value)
        {
            Node* newNode = new Node();
            newNode->data = value;
            newNode->nextElement = head;
            
            head = newNode;
            cout << value << " Inserted." << endl;
        }

        // search if element exists in the list
        bool search(int value)
        {
            Node* temp = head;
            while (temp != nullptr)
            {
                if (temp->data == value)
                {
                    return true;
                }
                temp = temp->nextElement;
            }
            delete temp;
            return false;

        }
};

int main()
{
    LinkedList list;
    srand(time(NULL));

    int num = 0;
    for (int i = 1; i < 10; i++)
    {
        num = rand() % 10 + 1;
        list.insertAtHead(num);
        list.printList();
    }

    bool exist = false;   // exist it true only if element exists
    for(int j = 0; j < 5; j++) {
        num = rand() % 10 + 1; 
        exist = list.search(num);   // Calling search function
        if(exist) {     // only true if element exists
            cout << num << " exists!" << endl;
        }
        else {
            cout << num << " does not exist!" << endl;
        }
    }
}